mycelium/src/tun/darwin.rs

```diff
@@ -174,6 +172,53 @@ fn validate_utun_name(input: &str) -> bool {
         .is_ok()
 }
 
+/// Find an available utun interface name
+fn find_available_utun_name(preferred_name: &str) -> Result<String, io::Error> {
+    let interfaces = datalink::interfaces();
+    let utun_interfaces: Vec<_> = interfaces
+        .iter()
+        .filter(|iface| iface.name.starts_with("utun"))
+        .collect();
+
+    if !preferred_name.is_empty() {
+        if !validate_utun_name(preferred_name) {
+            error!("Invalid TUN name: {preferred_name}. Name must start with 'utun' followed by digits");
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "Invalid TUN name",
+            ));
+        }
+        if interfaces.iter().any(|iface| iface.name == preferred_name) {
+            error!("TUN name {preferred_name} already in use");
+            return Err(io::Error::new(
+                io::ErrorKind::AlreadyExists,
+                "TUN name already in use",
+            ));
+        }
+        return Ok(preferred_name.to_string());
+    }
+
+    let max_utun_number = utun_interfaces
+        .iter()
+        .filter_map(|iface| iface.name[4..].parse::<u64>().ok())
+        .max()
+        .unwrap_or(0);
+
+    let new_utun_number = max_utun_number + 1;
+    let new_utun_name = format!("utun{}", new_utun_number);
+
+    if validate_utun_name(&new_utun_name) {
```

Comment:
do we still need to validate the name we created?

Reply:
In the highly unlikely case that a user has an interface with the name `utun99999999999`, adding +1 would cause the TUN name to exceed 15 char length, thus validating seems plausible IMO